libdns.go

```diff
@@ -1,394 +1,212 @@
+	// use.
 	AppendRecords(ctx context.Context, zone string, recs []Record) ([]Record, error)
 }
 
 // [RecordSetter] can set new or update existing records in a DNS zone.
-//
-// “SetRecords” updates the zone so that the records described in the input are
-// reflected in the output. It may create or overwrite records or—depending on
-// the record type—delete records to maintain parity with the input. No other
-// records are affected. It returns the records which were set.
-//
-// For any (name, type) pair in the input, “SetRecords” ensures that the only
-// records in the output zone with that (name, type) pair are those that were
-// provided in the input.
-//
-// In RFC 7719 terms, “SetRecords” appends, modifies, or deletes records in the
-// zone so that for each RRset in the input, the records provided in the input
-// are the only members of their RRset in the output zone.
-//
-// Implementations may decide whether or not to support DNSSEC-related records
-// in calls to “SetRecords”, but should document their decision. Note that the
-// decision to support DNSSEC records in “SetRecords” is independent of the
-// decision to support them in [libdns.RecordGetter.GetRecords], so end-users
-// should not blindly call “SetRecords” on the output of
-// [libdns.RecordGetter.GetRecords].
-//
-// Implementations must honor context cancellation and be safe for concurrent
-// use.
-//
-// # Examples
-//
-// Example 1:
-//
-//	;; Original zone
-//	example.com. 3600 IN A   192.0.2.1
-//	example.com. 3600 IN A   192.0.2.2
-//	example.com. 3600 IN TXT "hello world"
-//
-//	;; Input
-//	example.com. 3600 IN A   192.0.2.3
-//
-//	;; Resultant zone
-//	example.com. 3600 IN A   192.0.2.3
-//	example.com. 3600 IN TXT "hello world"
-//
-// Example 2:
-//
-//	;; Original zone
-//	a.example.com. 3600 IN AAAA 2001:db8::1
-//	a.example.com. 3600 IN AAAA 2001:db8::2
-//	b.example.com. 3600 IN AAAA 2001:db8::3
-//	b.example.com. 3600 IN AAAA 2001:db8::4
-//
-//	;; Input
-//	a.example.com. 3600 IN AAAA 2001:db8::1
-//	a.example.com. 3600 IN AAAA 2001:db8::2
-//	a.example.com. 3600 IN AAAA 2001:db8::5
-//
-//	;; Resultant zone
-//	a.example.com. 3600 IN AAAA 2001:db8::1
-//	a.example.com. 3600 IN AAAA 2001:db8::2
-//	a.example.com. 3600 IN AAAA 2001:db8::5
-//	b.example.com. 3600 IN AAAA 2001:db8::3
-//	b.example.com. 3600 IN AAAA 2001:db8::4
 type RecordSetter interface {
+	// SetRecords updates the zone so that the records described in the input are
+	// reflected in the output. It may create or update records or—depending on
+	// the record type—delete records to maintain parity with the input. No other
+	// records are affected. It returns the records which were set.
+	//
+	// For any (name, type) pair in the input, SetRecords ensures that the only
+	// records in the output zone with that (name, type) pair are those that were
+	// provided in the input.
+	//
+	// In RFC 7719 terms, SetRecords appends, modifies, or deletes records in the
```

Comment:
Not really related, but this could be a good moment to update the reference to  RFC 9499. The updates have been: 7719 -> 8499 -> 9499. See https://datatracker.ietf.org/doc/html/rfc9499 and follow the "Obsoletes" references in the pane on the right. The RFC is referenced earlier in this file too, and perhaps other places.

Reply:
Good point... yeah I'd be okay with updating it.

Reply:
Oh wow, I don't know how I missed that chain of “obsoleted by”s; good catch. Plus, RFC 9499 is way easier to understand, so I'll definitely update this.